packages/mui-base/src/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,352 @@
+    expect(checkbox).to.have.attribute('data-readonly', 'true');
+    expect(checkbox).to.have.attribute('data-required', 'true');
+
+    expect(indicator).to.have.attribute('data-state', 'checked');
+    expect(indicator).to.have.attribute('data-disabled', 'true');
+    expect(indicator).to.have.attribute('data-readonly', 'true');
+    expect(indicator).to.have.attribute('data-required', 'true');
+  });
+
+  it('should set the name attribute on the input', () => {
+    const { container } = render(<Checkbox name="checkbox-name" />);
+    const input = container.querySelector('input[type="checkbox"]')! as HTMLInputElement;
+
+    expect(input).to.have.attribute('name', 'checkbox-name');
+  });
+
+  describe('form handling', () => {
+    it('should toggle the checkbox when a parent label is clicked', function test() {
+      // Clicking the label causes unrelated browser tests to fail.
+      if (!isJSDOM) {
+        this.skip();
+      }
+
+      const { getByTestId, getAllByRole } = render(
+        <label data-testid="label">
+          <Checkbox />
+          Toggle
+        </label>,
+      );
+
+      const [checkbox] = getAllByRole('checkbox');
+      const label = getByTestId('label');
+
+      expect(checkbox).to.have.attribute('aria-checked', 'false');
+
+      act(() => {
+        label.click();
+      });
+
+      expect(checkbox).to.have.attribute('aria-checked', 'true');
+    });
+
+    it('should toggle the checkbox when a linked label is clicked', function test() {
+      // Clicking the label causes unrelated browser tests to fail.
+      if (!isJSDOM) {
+        this.skip();
+      }
+
+      const { getByTestId, getAllByRole } = render(
+        <div>
+          <label htmlFor="test-checkbox" data-testid="label">
+            Toggle
+          </label>
+          <Checkbox id="test-checkbox" />
+        </div>,
+      );
+
+      const [checkbox] = getAllByRole('checkbox');
+      const label = getByTestId('label');
+
+      expect(checkbox).to.have.attribute('aria-checked', 'false');
+
+      act(() => {
+        label.click();
+      });
+
+      expect(checkbox).to.have.attribute('aria-checked', 'true');
+    });
+  });
+
+  it('should include the checkbox value in the form submission', function test() {
+    if (isJSDOM) {
+      // FormData is not available in JSDOM
+      this.skip();
+    }
+
+    let stringifiedFormData = '';
+
+    const { getAllByRole, getByRole } = render(
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          const formData = new FormData(event.currentTarget);
+          stringifiedFormData = new URLSearchParams(formData as any).toString();
+        }}
+      >
+        <Checkbox name="test-checkbox" />
+        <button type="submit">Submit</button>
+      </form>,
+    );
+
+    const [checkbox] = getAllByRole('checkbox');
+    const submitButton = getByRole('button')!;
+
+    submitButton.click();
+
+    expect(stringifiedFormData).to.equal('');
```

Comment:
The Switch renders a hidden input with the same `name` when the checkbox is not checked to always send something to the server. I think it's worth having here as well. I remeber often having to implement this manually during my full-stack days.